File: frontend/src/services/admin/dashboard.service.ts
import { axiosClient } from "@/lib/api/axios";

import type {
  DashboardResponse,
  DashboardStatisticsResponse,
  RevenueReportResponse,
  UserReportResponse,
  CourseReportResponse,
} from "@/lib/admin/dashboard.types";

const PREFIX = "/admin";

export const adminDashboardService = {
  /* ================= DASHBOARD ================= */
  getDashboard: async (period: string): Promise<DashboardResponse> => {
    const res = await axiosClient.get(`${PREFIX}/dashboard`, {
      params: { period },
    });
    return res.data.data;
  },

  /* ================= STATISTICS ================= */
  getStatistics: async (
    period: string
  ): Promise<DashboardStatisticsResponse> => {
    const res = await axiosClient.get(`${PREFIX}/statistics`, {
      params: { period },
    });
    return res.data.data;
  },

  /* ================= REPORTS ================= */
  getRevenueReport: async (
    period: string
  ): Promise<RevenueReportResponse> => {
    const res = await axiosClient.get(`${PREFIX}/reports/revenue`, {
      params: { period },
    });
    return res.data.data;
  },

  getUserReport: async (
    period: string
  ): Promise<UserReportResponse> => {
    const res = await axiosClient.get(`${PREFIX}/reports/users`, {
      params: { period },
    });
    return res.data.data;
  },

  getCourseReport: async (
    period: string
  ): Promise<CourseReportResponse> => {
    const res = await axiosClient.get(`${PREFIX}/reports/courses`, {
      params: { period },
    });
    return res.data.data;
  },
};

File: backend/lms/src/main/java/vn/uit/lms/service/admin/DashboardService.java
package vn.uit.lms.service.admin;

import vn.uit.lms.shared.dto.response.admin.CourseReportResponse;
import vn.uit.lms.shared.dto.response.admin.DashboardResponse;
import vn.uit.lms.shared.dto.response.admin.DashboardStatisticsResponse;
import vn.uit.lms.shared.dto.response.admin.RevenueReportResponse;
import vn.uit.lms.shared.dto.response.admin.UserReportResponse;

public interface DashboardService {

    DashboardResponse getDashboard(String period);

    DashboardStatisticsResponse getStatistics(String period);

    RevenueReportResponse getRevenueReport(String period);

    UserReportResponse getUserReport(String period);

    CourseReportResponse getCourseReport(String period);
}

